### ISSUE TITLE
Refactor input component for accessibility

### ISSUE THREAD

**Issue by SeamusLeonardHPE** [grommet/hpe-design-system#3622]

**Comment by halocline** [comment]
I don't believe the core issue has to do with the design or implementation of the disabled inputs or buttons. WCAG states color contrast requirements do not apply to "inactive user interface components." Disabled elements are inactive. I believe the issue is applying disabled components incorrectly. My understanding of the context was that the design should have been using readonly inputs instead of disabled.

[WCAG 2.1 - Contrast Minimum 1.4.3](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum) - "Text or images of text that are part of an inactive [user interface component](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum#dfn-user-interface-component), ... have no contrast requirement."

[WCAG 2.1 - Non-text contrast ration 1.4.11](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast) - "Text or images of text that are part of an inactive [user interface component](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum#dfn-user-interface-component), that are [pure decoration](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum#dfn-pure-decoration), that are not visible to anyone, or that are part of a picture that contains significant other visual content, have no contrast requirement."

I think it is worthwhile to time-box a research spike to see if there are improvements which can be made to our disabled states and/or guidance (e.g. tweaks, use of non-color such as cursor=not-allowed, etc.), but want to be clear that the issue is not that the Design System disabled component states are inaccessible, rather, in some contexts, implementors are applying the disabled states incorrectly resulting in an inaccessible experience.

**Comment by SeamusLeonardHPE** [comment]
@halocline agree with your points.
Think this is an opportunity for us to improve guidance/education in figma via component variant metadata, and to flag potential accessibility issues in the documentation.

**Comment by halocline** [comment]
With the ReadOnly input work and clarification that this is likely a guidance related issue, does this work need to be rescoped? If so, can we redefine the deliverables?
